### PR TITLE
doc/dev: document nested subsets in teuthology

### PR DESCRIPTION
**Pull request by batrick**
See also: https://github.com/ceph/teuthology/pull/1704

Signed-off-by: Patrick Donnelly <pdonnell@redhat.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by batrick** [comment]
https://ceph--44967.org.readthedocs.build/en/44967/dev/developer_guide/testing_integration_tests/tests-integration-testing-teuthology-intro/#nested-subsets

File: doc/dev/developer_guide/testing_integration_tests/tests-integration-testing-teuthology-intro.rst [hunk 1]
```diff
@@ -384,6 +384,34 @@ An individual test from the `ceph-deploy suite`_ can be run by adding the
...
+
+Note the numerator is not specified as with the ``--subset`` option as there is
+no meaningful way to express this when there could be several layers of
+nesting.  Instead, a random subset is selected (1 of 2 in our example). The
+choice is based off the random seed (``--seed``) used for the scheduling.
+Remember that seed is saved in the results so that a ``--rerun`` of failed
+tests will still preserve the correct numerator (subset of subsets).
+
+You can disable nested subsets using the ``--no-nested-subset`` argument to
+``teuthology-sute``.
```
Comment (zmc) [inline doc/dev/developer_guide/testing_integration_tests/tests-integration-testing-teuthology-intro.rst:413]: typo

Comment (zmc) [inline doc/dev/developer_guide/testing_integration_tests/tests-integration-testing-teuthology-intro.rst:413]:
```suggestion
``teuthology-suite``.
```

**Review by zmc — CHANGES_REQUESTED** [review]

File: doc/dev/developer_guide/testing_integration_tests/tests-integration-testing-teuthology-intro.rst (hunk 1 above)
Comment (batrick) [reply doc/dev/developer_guide/testing_integration_tests/tests-integration-testing-teuthology-intro.rst]: Thanks, fixed!

**Review by zmc — APPROVED** [review]

**Pull request merged**